doc/level0.rst

```diff
@@ -92,26 +92,150 @@ To solve this problem in a better way, bi-directional communication
+e-mail message called the Autocrypt Setup Message.  An
+already-configured Autocrypt client can generate an Autocrypt Setup
+Message, and send it to itself.  A not-yet-configured Autocrypt client
+(a new client in a multi-device case, or recovering from device
+failure or loss) can import the Autocrypt Setup Message and recover
+the ability to read existing messages.
+
+An Autocrypt Setup Message is protected with a strong Setup Code.
+
+Message Structure
+++++++
+
+The Autocrypt Setup Message itself is an e-mail message with a
+specific format, which contains a payload protected by the setup code.
+
+- Both the To and From headers MUST be the address of the user.
+- The Autocrypt Setup Message MUST have a `multipart/mixed` structure,
+  with the first part a `text/plain` or `text/html` part that provides
+  a human-readable description to the user about the purpose of the
+  message.
+- The "payload" of the Autocrypt setup message is a MIME part that is
+  a direct child of the root part, with a Content-Type of
+  `application/pgp-key-backup`.  There MUST be exactly one payload
+  part of this Content-Type in the message.
+- The payload MUST contain a single ASCII-armored block of OpenPGP
+  symmetrically encrypted data, and MAY include other text above or
+  below the ASCII-armored data, which MUST be ignored while
+  processing.
+- Decrypting the payload should produce an ASCII-armored OpenPGP
+  transferable secret key.
+- The encryption algorithm used MUST be at least as strong as 128-bit
+  symmetric encryption.  It SHOULD be AES-128.  The passphrase MUST be
+  the Setup Code (see below), used under OpenPGP's salted+iterated S2K
+  algorithm.
+
+The payload MIME part MAY optionally use the OpenPGP Transfer Format
+(see :ref:`openpgp-transfer-format`) for improved usability.
+
+Setup Code
++++++
+
+The setup code MUST be generated by the implementation itself using a
+CSPRNG, and presented directly to the user for safekeeping. It MUST
+NOT be included in the Autocrypt Setup Message, or otherwise
+transmitted over e-mail. Its format SHOULD be 24 uppercase
+alphanumeric characters, divided into six blocks of four, separated by
+dashes. The dashes are part of the secret code. This format holds 124
+bits of data, it is designed to be unambiguous, pronounceable, and
+split into blocks that can be easily kept in short term memory. For
+instance:
+
+``AB1D-E2GH-IJK3-4NOP-Q5ST-XYZ6``
+
+When this form of Setup Code is used, the ASCII-armored block in the
+payload SHOULD include an OpenPGP Armor header called
+"Autocrypt-Setup-Message" with a value of "v1", to indicate to
+receiving implementations the format of the passphrase.  This is a
+useful indicator at message import time.
+
+Setup Message Creation
+++++++
+
+An Autocrypt client MUST NOT create an Autocrypt Setup Message without
+explicit user interaction.  When the user takes this action for a
+specific account, the client:
```

Comment:
patrick, bjoern, oliver, holger think it's worthwhile to allow for automatic recognition of an incoming setup message from another device.  it's probably helpful to have an ``Autocrypt-Setup`` header which allows for trivial and more precise recognition of setup messages (otherwise we might need to look at all bodies of all mails which have ``From: X To: X`` settings).  Also it's possible that a user has different identities/aliases configured on the sending and the receiving device.